src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -359,6 +359,34 @@ public string DefaultPythonEngine
         /// </summary>
         private static string defaultPythonEngine;
 
+        internal event Func<string> RequestUserDataFolder;
+        internal string OnRequestUserDataFolder()
+        {
+            return RequestUserDataFolder?.Invoke();
+        }
+
+        private string selectedPackagePathForInstall;
+        // TODO: Add this to IPreferences in Dynamo 3.0
+        /// <summary>
+        /// Currently selected package path where all packages downloaded from the Package Manager
+        /// will be installed. The default package path for install is the user data directory
+        /// currently used by the Dynamo environment.
+        /// </summary>
+        public string SelectedPackagePathForInstall {
+            get
+            {
+                if (string.IsNullOrEmpty(selectedPackagePathForInstall))
+                {
+                    selectedPackagePathForInstall = OnRequestUserDataFolder();
```

Comment:
is it safe if this is null (lets so there is no handler)

Reply:
Same comment as above. To answer your questions about null and other subscribers, there might be unknown issues but since there is at least one subscriber, I don't see a case where it will be null.